### ISSUE TITLE
pod install 0.95.0 failed

### ISSUE THREAD

**Issue by goreycn** [realm/realm-cocoa#2446]
Try to install realm with Cocoapods, I got the message below

```c
Installing Realm (0.95.0)
[!] /bin/bash -c
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.92.1.1
Downloading core failed. Please try again once you have an Internet connection.
```

and pod version:
```c
pod --version
0.38.2
```

**Comment by segiddins** [comment]
Can you try again, ensuring that you have an active internet connection while installing?

**Comment by codesourse** [comment]
Installing Realm (0.94.1)
[!] /bin/bash -c
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.92.1.1
Downloading core failed. Please try again once you have an Internet connection.

**Comment by codesourse** [comment]
Installing Realm (0.95.0)
[!] /bin/bash -c
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.92.1.1
Downloading core failed. Please try again once you have an Internet connection.

**Comment by codesourse** [comment]
I meet it, too

**Comment by segiddins** [comment]
@jpsim is it possible your S3 configuration changes could've caused this?

**Comment by codesourse** [comment]
I installing others is ok.
 
Downloading dependencies
Installing AFNetworking (2.5.4)
Installing ASProgressPopUpView (0.8)
Installing Base64nl (1.2)
Installing FCUUID (1.1.4)
Installing GZIP (1.0.3)
Installing IGDigest (1.1.2)
Installing Mantle (2.0.4)
Installing Realm (0.95.0)
[!] /bin/bash -c
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.92.1.1
Downloading core failed. Please try again once you have an Internet connection.

**Comment by segiddins** [comment]
This is strange, since I'm not having any trouble downloading the core zip myself

**Comment by jpsim** [comment]
> @jpsim is it possible your S3 configuration changes could've caused this?

That's the most likely option I can think of too. @codesourse would it be possible for you to share your public IP address with us? You can email it to us privately to help@realm.io if you prefer. We recently had to implement some IP filtering to prevent Realm from being downloaded from countries that we aren't allowed to export to, and I get the feeling that this filtering is affecting you.

**Comment by bdash** [comment]
I can reproduce the error myself 9 times out of 10. Running the curl step manually, with arguments tweaked to get decent errors, I see that I'm getting `HTTP/1.1 403 Forbidden`.

**Comment by codesourse** [comment]
@jpsim i sent your email.

**Comment by goreycn** [comment]
>We recently had to implement some IP filtering to prevent Realm from being downloaded from countries that we aren't allowed to export to, and I get the feeling that this filtering is affecting you.

That's the point ,  so I have to give it up (cocoapod)?  may be I should try the realm.framework again.

**Comment by haitham-aboshahba-sociomantic** [comment]
I have the exact same error with Realm 0.95, internet connection is active, `cocoapods 0.38.2`

**Comment by Padarom** [comment]
Can confirm this (from Germany). Have been wondering why I'm getting a message such as `Version <html><head><title>403 Forbidden</title></head>....</html> is now available ....` and can't update now. Before looking deeper into this issue, I have followed another recommendation to delete a couple of folders (including `Pods/RealmCocoa`), so my current installation is gone now.

It's not this IP address alone, I've tried from work and from home and it's `403 Forbidden` for both.

**Comment by haitham-aboshahba-sociomantic** [comment]
@Padarom  exactly, i'm trying from Berlin :) and my installation is gone as i deleted the `Pods` and `xxxx.xcworkspace`

**Comment by MickeyHub** [comment]
Beacause of bad net env.
the prepare command Realm provide is `sh build.sh cocoapod-setup`,  mainly code is download realm-core, download realm-core if $TMPDIR/core_bin/core-${REALM_CORE_VERSION}.tar.bz2 file not exists. so if you downloads always failed, you can download realm-core directly, and then backup it, when you need pod install realm, you can mv it to $TMPDIR/core_bin/core-${REALM_CORE_VERSION}.tar.bz2.  

${REALM_CORE_VERSION} is core version , such as `0.92.1.1`

**Comment by Padarom** [comment]
No we can not. There's even a `403 Forbidden` when manually going on `static.realm.io` (which is where it's hosted), no matter which file we try to access. So there's no chance manually downloading without using a VPN or something the like.

**Comment by MickeyHub** [comment]
I put last realm-core 0.92.1.1 to my dropbox, 
https://www.dropbox.com/s/r0w7u1ox5n4i3ms/core-0.92.1.1.tar.bz2?dl=0

**Comment by haitham-aboshahba-sociomantic** [comment]
great , Thank (Y)

**Comment by haitham-aboshahba-sociomantic** [comment]
i confirm it's working now after downloading the file from @MickeyHub 's dropbox and placing into `$TMPDIR/core_bin/`

thanks again

**Comment by Padarom** [comment]
Worked like a charm, thanks :+1: At least I'm able to get working again for now :)

Though the issue should still be looked into for the future

**Comment by MickeyHub** [comment]
almost always failed for China's net env.😂

**Comment by jpsim** [comment]
Sorry about the trouble, we've disabled IP filtering on AWS S3 for the time being so you should now be able to download core (and other assets served from static.realm.io).

**Comment by austinkettner** [comment]
I am experiencing this exact issue right now targetting 0.95.0

**Comment by jpsim** [comment]
@austinkettner we're experiencing issues with S3 since 9am Pacific Time this morning. I'll post again here once we've resolved the issues.

**Comment by austinkettner** [comment]
Thanks @jpsim, much appreciated!

**Comment by jpsim** [comment]
Ok, now resolved. It may take a few minutes for these changes to propagate. Thanks for your patience.

**Comment by bnadim** [comment]
Hi, I am experiencing the same issue from France trying to get the branch swift 2.0

**Comment by jpsim** [comment]
@bnadim that sounds like a different issue, please open a new ticket at https://github.com/realm/realm-cocoa/issues/new

**Comment by luohui8891** [comment]
Still got following issue:
```
[!] /bin/bash -c
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.92.2
Downloading core failed. Please try again once you have an Internet connection.
```

**Comment by jpsim** [comment]
Looks like Realm's core binaries couldn't be downloaded, as the message indicates. If you generally have difficulty downloading large assets from AWS S3 hosted in the US, that could explain the issue. Only advice I can give is to try again.

**Comment by luohui8891** [comment]
Thanks @jpsim . And yes, the core has been download after 10+ times.

**Comment by MickeyHub** [comment]
Oh,no. My attemps times has gt 10, result is the same as before.

**Comment by beeender** [comment]
@luohui8891 @MickeyHub would you please send me your contact info(email, wechat or qq) to mc@realm.io? I am in China now, maybe I can help to identify the problem.

**Comment by MickeyHub** [comment]
My qq: 791331313 , verification-info:15651787836

**Comment by luohui8891** [comment]
@beeender email sent.

**Comment by luohui8891** [comment]
@MickeyHub hey guy, just FYI, I have tried again via US VPN, it works.

**Comment by neugartf** [comment]
Same here, tried it several times. No success.
```
[!] /bin/bash -c 
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.94.0
Downloading core failed. Please try again once you have an Internet connection.
```

Is there any ip address or url I can verify with that this is in fact caused by aws s3?
Thanks!

**Comment by neugartf** [comment]
thanks @MickeyHub , that worked! Although it's important to rename the downloaded file to "core-VERSION.tar.bz2".

**Comment by MickeyHub** [comment]
@luohui8891  ,Thank you, I think it'll be OK.
@neugartf , I updated dropbox share link to latest realm-core(0.94.0, tag:v0.96.0)
[download latest core file](https://www.dropbox.com/s/8etwdpyl9ynp81v/core-0.94.0.tar.bz2?dl=0)

**Comment by losingle** [comment]
I meet it, too
```
 > Running prepare command
   $ /bin/bash -c  set -e sh build.sh cocoapods-setup
   core is not a symlink. Deleting...
   Downloading dependency: core 0.94.0
   Downloading core failed. Please try again once you have an Internet connection.
[!] /bin/bash -c
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.94.0
Downloading core failed. Please try again once you have an Internet connection.
```

**Comment by beeender** [comment]
@losingle What is the ISP you are using? China Telecom?

**Comment by losingle** [comment]
@beeender China Telecom over Japan VPN

**Comment by beeender** [comment]
@losingle What is yours VPN provider? Or we can discuss through email: mc@realm.io

**Comment by MickeyHub** [comment]
ditto

**Comment by hengchengfei** [comment]
directly download the latest version core-0.94.0.tar.bz2.
put it to the folder $TMPDIR/core_bin.

![2015-10-26 4 40 08](https://cloud.githubusercontent.com/assets/3182964/10724745/c04a98ca-7c01-11e5-8e4e-3ced65534612.png)

**Comment by allensu212** [comment]
have the same issue here:

[!] /bin/bash -c 
set -e
sh build.sh cocoapods-setup

core is not a symlink. Deleting...
Downloading dependency: core 0.95.5
Downloading core failed. Please try again once you have an Internet connection.

@jpsim ,does the team include Taiwan as one of countries that aren't allowed to export to? Thank you

**Comment by jpsim** [comment]
> @jpsim ,does Taiwan is one of countries that is affected by IP filtering? Thank you

No, only the countries listed as blocked in our [README](https://github.com/realm/realm-cocoa#license).

**Comment by ytrugg** [comment]
I have the same issue here from Holland. I'm trying to download realm core 0.95.5.

**Comment by cookeri6** [comment]
Having this issue from the US. Any progress been made on this?

**Comment by mrackwitz** [comment]
We're tracking this issue now by #2713, please take a look at this proposed workaround there and whether your error matches the output shown there, when trying to download the latest version of Realm. If that shouldn't fix it, please open up a new issue instead.

**Comment by zhangxistudy11** [comment]
@MickeyHub  You  are  right.I  meet  this  issue  frequently in China.

**Comment by bitraven** [comment]
I had this problem today after updating Cocoapods from 0.39.0 to 1.0.1 then to 1.1.0.rc.2

prior to the 1.0.1 upgrade, pod install would properly install RealmSwift, but after, I got the error:

[!] /bin/bash -c 
set -e
sh build.sh cocoapods-setup

I downgraded back to 0.39.0 and ran pod install again, and received the following (new) error:

xcrun: error: active developer path ("/Applications/Xcode-beta.app/Contents/Developer") does not exist, use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools (or see `man xcode-select`)
Downloading dependencies

I deleted Xcode-beta from my system a few days/weeks ago when GM8 hit, so running xcode-select -r to reset to the default command line tools path resolved that error on 0.39.0

on a whim, I re-upgraded to 1.0.1 and re-ran pod install.

This time, no issues, everything installed just fine.

Not sure if it's the same issue as what other (US) people have seen, but it might be worth looking in to if you're hitting this error.  (I did not add this to #2713 because I'm not in China ...)
